### PR TITLE
Implemented Runtime Permissions Properly.

### PR DESCRIPTION
**Pull request by CoderVishalSehgal**
## Fixed the Runtime Permissions Flow. Please check it out.
@thedevelopersanjeev #574 

travelmate-runtimepermissiondemo-codervishalsehgal

### CODE REVIEW

**Comment by Swati4star** [comment]
why have you added bg_login.png file. it is not being used anywhere

File: Android/app/build.gradle [hunk 1]
```diff
@@ -30,6 +30,7 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
```
Comment (Swati4star) [inline Android/app/build.gradle]: remove this file from commit

File: Android/app/src/androidTest/java/io/github/project_travel_mate/ApplicationTest.java [hunk 2]
```diff
@@ -1,13 +0,0 @@
-package io.github.project_travel_mate;
```
Comment (Swati4star) [inline Android/app/src/androidTest/java/io/github/project_travel_mate/ApplicationTest.java]: why did you delete this?

File: Android/app/build.gradle [hunk 3]
```diff
@@ -99,8 +100,14 @@ dependencies {
 
     //Graph view
     implementation 'com.github.PhilJay:MPAndroidChart:v3.0.3'
+
+
+    androidTestImplementation('com.android.support.test.espresso:espresso-core:2.2.2', {
```
Comment (Swati4star) [inline Android/app/build.gradle]: why have you added this?

File: Android/app/src/androidTest/java/io/github/project_travel_mate/ExampleInstrumentedTest.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static junit.framework.TestCase.assertEquals;
+
+@RunWith(AndroidJUnit4.class)
+public class ExampleInstrumentedTest {
+    @Test
+    public void useAppContext() throws Exception {
```
Comment (Swati4star) [inline Android/app/src/androidTest/java/io/github/project_travel_mate/ExampleInstrumentedTest.java]: why is there need to add this test for this PR?

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 5]
```diff
@@ -408,7 +468,7 @@ void showProfileOrTrip(String data) {
                 Log.v("trip id", tripID + " ");
                 Trip trip = new Trip();
                 trip.setId(tripID);
-                Intent intent = MyTripInfoActivity.getStartIntent(MainActivity.this,  trip, false);
+                Intent intent = MyTripInfoActivity.getStartIntent(MainActivity.this, trip, false);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: why this change?

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java [hunk 6]
```diff
@@ -331,7 +337,9 @@ public void onResponse(Call call, final Response response) {
                                         arr.getJSONObject(i).getInt("facts_count"),
                                         R.color.sienna,
                                         getString(R.string.interest_know_more), getString(R.string.interest_weather),
-                                        getString(R.string.interest_fun_facts), getString(R.string.interest_trends)));
+                                        getString(R.string.
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java]: no need of this change

File: Android/app/src/androidTest/java/io/github/project_travel_mate/ApplicationTest.java (hunk 2 above)
Comment (CoderVishalSehgal) [reply Android/app/src/androidTest/java/io/github/project_travel_mate/ApplicationTest.java]: Because it was deprecated and gives me an error while building the project.

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java (hunk 5 above)
Comment (CoderVishalSehgal) [reply Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: I haven't changed this, I just pressed ctrl+alt+L to format the code. What's wrong with this, Can you please tell me?

File: Android/app/build.gradle (hunk 1 above)
Comment (CoderVishalSehgal) [reply Android/app/build.gradle]: Which file? Can you please explain it clearly?!

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java (hunk 6 above)
Comment (CoderVishalSehgal) [reply Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java]: It was maybe a mistake, I've fixed it.

File: Android/app/src/androidTest/java/io/github/project_travel_mate/ExampleInstrumentedTest.java (hunk 4 above)
Comment (CoderVishalSehgal) [reply Android/app/src/androidTest/java/io/github/project_travel_mate/ExampleInstrumentedTest.java]: Just replaced the deprecated Test which was there before.

File: Android/app/build.gradle (hunk 1 above)
Comment (Swati4star) [reply Android/app/build.gradle]: app/build.gradle

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java (hunk 5 above)
Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: We dont want any extra changes in PR which are unrelated to issue.

File: Android/app/src/androidTest/java/io/github/project_travel_mate/ApplicationTest.java (hunk 2 above)
Comment (Swati4star) [reply Android/app/src/androidTest/java/io/github/project_travel_mate/ApplicationTest.java]: There were no build errors before. there shouldnt be a need to remove this

File: Android/app/build.gradle (hunk 1 above)
Comment (Swati4star) [reply Android/app/build.gradle]: It's still here

File: Android/app/src/androidTest/java/io/github/project_travel_mate/ExampleInstrumentedTest.java (hunk 4 above)
Comment (Swati4star) [reply Android/app/src/androidTest/java/io/github/project_travel_mate/ExampleInstrumentedTest.java]: Dont do any other change thats needed apart for the issue. Revert it

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java (hunk 5 above)
Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: ??

**Comment by Swati4star** [comment]
@CoderVishalSehgal Are you working on this? There are still a number of issues that need to be addressed

**Comment by CoderVishalSehgal** [comment]
@Swati4star Reverted requested changes

File: Android/app/src/main/res/layout/layout_runtime_permissions_description.xml [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+                app:layout_constraintStart_toStartOf="@+id/imageView7"
+                app:layout_constraintTop_toBottomOf="@+id/textView7"
+                app:srcCompat="@drawable/ic_local_phone_black_24dp" />
+
+            <TextView
+                android:id="@+id/textView8"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="8dp"
+                android:text="we need this permission to directly call hotels directly from the app"
```
Comment (Swati4star) [inline Android/app/src/main/res/layout/layout_runtime_permissions_description.xml]: dont add constant strings in layout file. they should be in strings.xml file

**Comment by prabhakar267** [comment]
@CoderVishalSehgal Please update the PR.

**Comment by prabhakar267** [comment]
Closing this due to inactivity.
@CoderVishalSehgal reopen this if you update
